### DISCUSSION TITLE
Support Bambu Studio

### DISCUSSION THREAD

**Discussion by kens96** [NilsRo/OctoPrint-SlicerEstimator#69] [Ideas]
I have an Ender 3v2 running Klipper with OctoPrint. SlicerEstimater works well with Orca slicer, would it be possible to support Bambu Studio now that it has support for several 3rd party printers?

![Screenshot 2024-01-03 155646](https://github.com/NilsRo/OctoPrint-SlicerEstimator/assets/33166143/e7f28428-54c0-42cc-9133-531e6be2f92b)

**Comment by NilsRo** [comment]
Hi,

can you upload a Gcode file with time information in it (sometimes it has to be enabled manually). I can add the format then.

Cheers,
Nils

**Reply by kens96** [reply]
Hi Nils,
Thank you for getting back on this.
I have attached a GCODE generated on Bambu Studio, I had to rename it to a .txt as it didn't like the extension.

Thanks!
Rob

[BBS-Support_top_PLA_36m30s.txt](https://github.com/NilsRo/OctoPrint-SlicerEstimator/files/13823348/BBS-Support_top_PLA_36m30s.txt)

**Reply by NilsRo** [reply]
There is a total print time available only. This is not perfect, as I can estimate the remaining time from the percentage printed (rows processed in Gcode!) only but it is not a direct relationship. Is it possible in the slicer to activate something like https://marlinfw.org/docs/gcode/M073.html commands that displays the remaining time in the printer that I can use also for updating OctoPrint (even if that are not supported in the printer).

**Reply by kens96** [reply]
Hi Nils, 
Apologies if I am misunderstanding but I can see the M73 in the gcode both percentage and remaining time on a periodic bases?
![Screenshot 2024-01-03 231635](https://github.com/NilsRo/OctoPrint-SlicerEstimator/assets/33166143/a6920114-5293-4e51-9cea-d65336d013a4)

**Reply by NilsRo** [reply]
Checked this on the coach looking TV with new tablet...wrong search criteria. :-) Yes, should be easy to integrate. Will do that the next days

**Reply by kens96** [reply]
Thank you very much Nils, really appreciate your support with this.

**Reply by NilsRo** [reply]
You can switch in the setting of the plugin to the development version and try is everything is working fine with Bambu Studio. The test with the provided file looks fine.

BTW. the metadata file format is not supported by OctoPrint so the printtime is not shown in the file explorer and as a result OctoPrints inbuild analyser dumps. During printing I can update it so there it will be fine. Perhaps you can open a ticket at the Bambo Studio maintainer...config block is fine, header block not.

![image](https://github.com/NilsRo/OctoPrint-SlicerEstimator/assets/51311453/196ead63-84d8-43eb-8208-c2f293fd8e46)

**Reply by kens96** [reply]
Hi Nils
I just sliced a new file with the Dev branch and it is and I now have a green tick, it is working. Thank you very much!
![image](https://github.com/NilsRo/OctoPrint-SlicerEstimator/assets/33166143/341e987d-9b34-4cd4-8e37-7beee1f6c78c)

I did notice the print time missing in the explorer, I certainly will open a ticket with Bambu and see what they say. 
BTW, this is the repo https://github.com/bambulab/BambuStudio

**Reply by NilsRo** [reply]
You can switch back to the standard release channel and force an update in OctoPrint to get Bambu Studio Support in the official version. But actually there is no difference to the one you are using but you should avoid staying in the development channel.
